### ISSUE TITLE
Pre-Activation ResNet (ResNet-V2) implementation error leads to NaN loss

### ISSUE THREAD

**Issue by htwang14** [kuangliu/pytorch-cifar#137]
Hi Kuang, 

I think a batch normalization is missing after the last residual block and before [the pooling/fully connected layer](https://github.com/kuangliu/pytorch-cifar/blob/49b7aa97b0c12fe0d4054e670403a16b6b834ddd/models/preact_resnet.py#L91) in the current pre-activation resnet implementation.

There should be something like  `out = F.relu(self.bn(out), inplace=True)` inserted there, where `self.bn = nn.BatchNorm2d(512)` should be define in the `__init__` function, as done in other public implementations such as [this one](https://github.com/hysts/pytorch_resnet_preact/blob/c76a0825ddc0bc58d239d737815bca05a1df5e8d/resnet_preact.py#L271). 

The current Pre-activation resnet cannot converge on CIFAR10. I tried SGD optimizer with initial learning rate 0.1 on pre-act resnet18. The loss soon goes to NaN after the first several iterations.

Fixing the codes in my proposed way leads to good convergence. 

:-)

Thanks,
Haotao

**Comment by Zhidong-Gao** [comment]
It's maybe due to the fact that there are two consequent batch normalization layers.

You already have a batch normalization layer (self.bn1) after self.conv1. While in self.layer1, the self._make_layer() creates another batch normalization layer after self.bn1. Two consequent batch normalization layers easily lead NaN.

:-) Remove self.bn1 should work properly.
